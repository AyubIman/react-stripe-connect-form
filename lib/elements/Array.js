import React from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';

import { Textbox } from 'weav';

import Object from './Object';


import { convertProperLabelFromKey, findNewFontSize } from '../utils/label';


const Array = (props) => {
  const {
    label,
    list,
    keysNotToDisplay,
    specialLabels,
    fontSize,
    onValueChange,
    objectPath,
  } = props;

  return (
    <div>
      {
        (label) ? (
          <p
            style={{
              fontSize,
            }}
          >
            {convertProperLabelFromKey(label, specialLabels)}
          </p>
        ) : (null)
      }
      {
        list.map((item, itemIndex) => {
          // "example 1"
          if (
            _.isString(item) &&
            !_.includes(keysNotToDisplay, item)
          ) {
            return (
              <Textbox
                value={item}
                onChange={(e) => { onValueChange(objectPath, e.target.value); }}
              />
            );
          } else if (
            _.isArray(item) &&
            !_.includes(keysNotToDisplay, item)
          ) {
            return (
              <Array
                {...props}
                list={item}
                fontSize={findNewFontSize(fontSize)}
                objectPath={`${objectPath}[${itemIndex}]`}
              />
            );
          } else if (
            _.isObject(item) &&
            !_.includes(keysNotToDisplay, item)
          ) {
            return (
              <Object
                {...props}
                obj={item}
                fontSize={findNewFontSize(fontSize)}
                objectPath={`${objectPath}[${itemIndex}]`}
              />
            );
          }
          return null;
        })
      }
    </div>
  );
};

Array.propTypes = {
  label: PropTypes.string,
  list: PropTypes.arrayOf(PropTypes.any),
  keysNotToDisplay: PropTypes.arrayOf(PropTypes.string),
  // eslint-disable-next-line react/forbid-prop-types
  specialLabels: PropTypes.object,
  fontSize: PropTypes.number.isRequired,
  onValueChange: PropTypes.func.isRequired,
  objectPath: PropTypes.string,
};

Array.defaultProps = {
  label: null,
  list: [],
  keysNotToDisplay: [],
  specialLabels: {},
  objectPath: '',
};

export default Array;
