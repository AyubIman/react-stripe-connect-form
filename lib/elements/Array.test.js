/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';

import Array from './Array';

describe('Array', () => {
  test('Shallow render of the array. Basic render, no props.', () => {
    const wrapper = shallow(
      <Array
        onValueChange={() => {}}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  test('Shallow render of the array. Basic render with label.', () => {
    const wrapper = shallow(
      <Array
        onValueChange={() => {}}
        label="Foobar"
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
