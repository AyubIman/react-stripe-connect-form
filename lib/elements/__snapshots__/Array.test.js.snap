// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Array Shallow render of the array. Basic render with label. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Array
    fontSize={14}
    keysNotToDisplay={Array []}
    label="Foobar"
    list={Array []}
    objectPath=""
    onValueChange={[Function]}
    specialLabels={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          style={
            Object {
              "fontSize": 14,
            }
          }
        >
          FOOBAR
        </p>,
        Array [],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "FOOBAR",
          "style": Object {
            "fontSize": 14,
          },
        },
        "ref": null,
        "rendered": "FOOBAR",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            style={
              Object {
                "fontSize": 14,
              }
            }
          >
            FOOBAR
          </p>,
          Array [],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "FOOBAR",
            "style": Object {
              "fontSize": 14,
            },
          },
          "ref": null,
          "rendered": "FOOBAR",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Array Shallow render of the array. Basic render, no props. 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Array
    fontSize={14}
    keysNotToDisplay={Array []}
    label={null}
    list={Array []}
    objectPath=""
    onValueChange={[Function]}
    specialLabels={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        Array [],
      ],
    },
    "ref": null,
    "rendered": Array [
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          Array [],
        ],
      },
      "ref": null,
      "rendered": Array [
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
